q2_amr/types/_format.py

```diff
@@ -430,3 +430,215 @@ class CARDGeneAnnotationDirectoryFormat(MultiDirValidationMixin, model.Directory
     @gene.set_path_maker
     def gene_path_maker(self, sample_id):
         return "%s/gene_mapping_data.txt" % sample_id
+
+
+class CARDMAGsKmerAnalysisFormat(model.TextFileFormat):
+    def _validate(self, n_records=None):
+        header_exp = [
+            "ORF_ID",
+            "Contig",
+            "Cut_Off",
+            "Best_Hit_ARO",
+            "CARD*kmer Prediction",
+            "Taxonomic kmers",
+            "Genomic kmers",
+        ]
+
+        df = pd.read_csv(str(self), sep="\t")
+        header_obs = list(df.columns)
+        if not set(header_exp).issubset(set(header_obs)):
+            raise ValidationError(
+                "Header line does not match CARDMAGsKmerAnalysisFormat. Must contain"
+                "the following values: "
+                + ", ".join(header_exp)
+                + ".\n\nFound instead: "
+                + ", ".join(header_obs)
+            )
+
+    def _validate_(self, level):
+        self._validate()
+
+
+class CARDMAGsKmerAnalysisJSONFormat(model.TextFileFormat):
+    def _validate(self, n_records=None):
+        keys_exp = [
+            "ORF",
+            "contig",
+            "HSP",
+            "ARO_model",
+            "type_hit",
+            "#_of_kmers_in_sequence",
+            "#_of_AMR_kmers",
+            "taxonomic_info",
+            "genomic_info",
+        ]
+        with open(str(self)) as json_file:
+            dict = json.load(json_file)
+
+        keys_obs = list(next(iter(dict.values())).keys())
+
+        if keys_obs != keys_exp:
```

Comment:
If you don't want a specific order to the columns, consider using a set comparison to ensure that all expected keys are present without enforcing the order. For example, use `set(keys_obs) != set(keys_exp)`